cmd/crank/beta/validate/cmd.go

```diff
@@ -117,6 +118,11 @@ func (c *Cmd) Run(k *kong.Context, _ logging.Logger) error {
 		c.CacheDir = filepath.Join(currentPath, c.CacheDir)
 	}
 
+	if strings.HasPrefix(c.CacheDir, "~/") {
```

Comment:
As I mentioned in the issue, this only solves the base case to return the current user's home directory, which I admit is likely the most common by far.  It does not solve the issue for the case where a specific other-user's home directory is referenced, as in `--cache-dir=~otheruser/.cache`.  I looked around a little and there does not seem to be an official way to do this in go, all of the solutions I saw only reference the base case.

I think it's unfortunate that the CLI package prints the help message with the `=` sign instead of a space, and includes the quotes around the default value for the cache.  The combination of those two things makes it unlikely that the user will try `--cache-dir ~/.crossplane/cache` which works fine in most modern shells.

I wonder if it would make sense to change the default cache location to ~/.crossplane/cache so the user doesn't end up with a bunch of cache directories lying around, since that sort of defeats the purpose of having the cache in the first place.  That can be a separate issue/PR.

Given that the scope of this change is limited by available go solutions I think it's fine, it's just unfortunate that go hasn't chosen a better path for CLI argument handling.